numba/core/typeinfer.py

```diff
@@ -1207,6 +1207,8 @@ def check_var(name):
 
         try:
             retty = self.get_return_type(typdict)
+            typdict = utils.UniqueDict(
+                typdict, **{v.name: retty for v in self._get_return_vars()})
```

Comment:
Don't think this is needed in the try...except. Can you move this out in the `else` part for the `try...except...else`?